lib/core.ts

```diff
@@ -134,16 +134,24 @@ export interface CurrentOptions {
   code?: CodeOptions // NEW
 }
 
+type RegExpEngine = (pattern: string, u: string) => RegExpLike
+
+export interface RegExpLike {
+  test: (s: string) => boolean
+}
+
 export interface CodeOptions {
   es5?: boolean
   lines?: boolean
   optimize?: boolean | number
   formats?: Code // code to require (or construct) map of available formats - for standalone code
   source?: boolean
   process?: (code: string, schema?: SchemaEnv) => string
+  regExp?: RegExpEngine & {code: string} // add type
```

Comment:
I think code: string could be added to the type itself and to the docs. Possibly as optional, if typescript doesn’t not complain - it’s only needed for standalone code generation